applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -0,0 +1,856 @@
+{
+    if (rValues.size() != 1)
+        rValues.resize(1);
+    if (rVariable == TRAILING_EDGE)
+        rValues[0] = this->GetValue(TRAILING_EDGE);
+    else if (rVariable == KUTTA)
+        rValues[0] = this->GetValue(KUTTA);
+    else if (rVariable == WAKE)
+        rValues[0] = this->GetValue(WAKE);
+    else if (rVariable == ZERO_VELOCITY_CONDITION)
+        rValues[0] = this->GetValue(ZERO_VELOCITY_CONDITION);
+    else if (rVariable == TRAILING_EDGE_ELEMENT)
+        rValues[0] = this->GetValue(TRAILING_EDGE_ELEMENT);
+    else if (rVariable == DECOUPLED_TRAILING_EDGE_ELEMENT)
+        rValues[0] = this->GetValue(DECOUPLED_TRAILING_EDGE_ELEMENT);
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::GetValueOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector<array_1d<double, 3>>& rValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rValues.size() != 1)
+        rValues.resize(1);
+    if (rVariable == VELOCITY){
+        const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+        array_1d<double, 3> v(3, 0.0);
+        array_1d<double, Dim> vaux = PotentialFlowUtilities::ComputeVelocity<Dim, NumNodes>(*this);
+        for (unsigned int k = 0; k < Dim; k++)
+            v[k] = vaux[k] + free_stream_velocity[k];
+        rValues[0] = v;
+    }
+    else if (rVariable == PERTURBATION_VELOCITY)
+    {
+        array_1d<double, 3> v(3, 0.0);
+        array_1d<double, Dim> vaux = PotentialFlowUtilities::ComputeVelocity<Dim,NumNodes>(*this);
+        for (unsigned int k = 0; k < Dim; k++)
+            v[k] = vaux[k];
+        rValues[0] = v;
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Input and output
+
+template <int Dim, int NumNodes>
+std::string TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::Info() const
+{
+    std::stringstream buffer;
+    buffer << "TransonicPerturbationPotentialFlowElement #" << Id();
+    return buffer.str();
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << "TransonicPerturbationPotentialFlowElement #" << Id();
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::PrintData(std::ostream& rOStream) const
+{
+    pGetGeometry()->PrintData(rOStream);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Private functions
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::GetWakeDistances(
+    array_1d<double, NumNodes>& distances) const
+{
+    noalias(distances) = GetValue(WAKE_ELEMENTAL_DISTANCES);
```

Comment:
If this element is embedded (I didn't realise until I got here), it must be specified in the class name.

Reply:
All of the elements in this app embed the wake, we only indicate embedded if they also embed the geometry as for the elements developed by @marcnunezc

Reply:
OK. Now it's much clear. I agree that in this context it is redundant to specify that the wake is embedded.

Maybe it is a good idea to clarify this in the application `REAMDE.md`.

Reply:
Ok Ill include it in the README